java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/client/utils/ClientAuthenticationUtils.java

```diff
@@ -115,6 +115,16 @@ static KeyStore getKeyStoreInstance(String instance)
     return keyStore;
   }
 
+  @VisibleForTesting
+  static KeyStore getDefaultKeyStoreInstance(String password)
+      throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+    try (InputStream fileInputStream = getKeystoreInputStream()) {
+      KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+      keyStore.load(fileInputStream, password == null ? null : password.toCharArray());
```

Comment:
Does it need to load all cacerts files? Is there a way to restrict files needed to be loaded based on security concerns?

Reply:
Could you elaborate on the security concerns from loading all the cacerts files?

Reply:
Based on the current user workaround, there is no more question for my side https://github.com/apache/arrow/pull/39235#discussion_r1430566562

Reply:
It looks good to me. Maybe it would be better to name this method `geetDefaultTrustStore()` but that's not a big deal.